accern_xyme/accern_xyme.py

```diff
@@ -751,6 +753,52 @@ def get_settings(self) -> Dict[str, Any]:
         assert self._settings is not None
         return self._settings
 
+    def get_timing(
+            self,
+            blacklist: str = "wait_for_uri") -> Dict[str, TimingResult]:
+        nodes = self.get_nodes()
+        pipe_timing: Dict[str, TimingResult] = {}
+        node_timing: Dict[str, NodeTiming] = {}
+
+        def filter_blacklist(
+                node_time: List[Timing]) -> Iterator[Timing]:
+            for k, v in enumerate(node_time):
+                if v.get("name") != blacklist:
+                    yield v
+        for node_ix, node in enumerate(nodes):
+            node_time = self.get_node(node).get_timing()
+            node_times: List[Timing] = [
+                time for time in filter_blacklist(node_time)]
+            node_time_dicts: Dict[str, NodeTiming] = {}
+            for pos, cur in enumerate(node_times):
+                node_name: str = self.get_node(node).get_node_def()["name"]
+                node_sums: float = cur.get("total")
+                node_ids = self.get_node(node).get_id()
+                length = len(node_time)
+                node_obj = node_time_dicts.get(node_ids, {
+                    "node_name": node_name,
+                    "node_total": 0.0,
+                    "node_avg": 0.0,
+                    "fns": node_time,
+                })
+                node_obj["node_total"] += node_sums
+                node_obj["node_avg"] += node_sums / length
```

Comment:
you cannot sum up averages -- you need to keep track of the total sum and total length and then use those to compute the average